ui/apps/platform/src/Containers/Clusters/ClustersTable.tsx

```diff
@@ -0,0 +1,246 @@
+                                                columnIndex: 4,
+                                            }}
+                                        >
+                                            {/* TODO: needs update for upgrade */}
+                                            <SensorUpgrade
+                                                upgradeStatus={clusterInfo.status?.upgradeStatus}
+                                                centralVersion={metadata.version}
+                                                sensorVersion={clusterInfo.status?.sensorVersion}
+                                                isList
+                                            />
+                                        </Td>
+                                        <Td dataLabel="Credential expiration">
+                                            {/* TODO: needs update for upgrade */}
+                                            <CredentialExpiration
+                                                certExpiryStatus={
+                                                    clusterInfo.status
+                                                        ?.certExpiryStatus as CertExpiryStatus
+                                                }
+                                                autoRefreshEnabled={clusterInfo.sensorCapabilities?.includes(
+                                                    'SecuredClusterCertificatesRefresh'
+                                                )}
+                                                isList
+                                            />
+                                        </Td>
+                                        <Td dataLabel="Cluster deletion">
+                                            <ClusterDeletion
+                                                clusterRetentionInfo={
+                                                    clusterIdToRetentionInfo[clusterId] ?? null
+                                                }
+                                            />
+                                        </Td>
+                                        <Td isActionCell>
+                                            <ActionsColumn
+                                                items={[
+                                                    {
+                                                        title: 'Delete cluster',
+                                                        onClick: (event) => {
+                                                            onDeleteCluster(clusterInfo)(event);
+                                                        },
+                                                    },
+                                                ]}
+                                            />
+                                        </Td>
+                                    </Tr>
+                                    {isCellExpanded(clusterId, COL.STATUS) && (
```

Comment:
Since both sections use the same `Tr`, `Td`, and `ExpandableRowContent` structure, we could refactor to reuse the shared render logic and reduce duplication.